src/bridges/mean/MeanBridge.sol

```diff
@@ -95,6 +96,53 @@ contract MeanBridge is BridgeBase, Ownable2Step {
         return (0, 0, true);
     }
 
+    /**
+     * @notice A function which will allow the users to close their DCA positions on Mean Finance
+     * @param _outputAssetA - ETH or ERC20 token that the position had swapped funds into
+     * @param _outputAssetB - ETH or ERC20 token that had been deposited by the user
+     * @param _interactionNonce - Unique identifier for this DeFi interaction
+     * @return _outputValueA - The amount of swapped funds
+     * @return _outputValueB - The amount of unswapped funds
+     * @return _interactionComplete - This will always be true
+     */
+    function finalise(
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata _outputAssetB,
+        uint256 _interactionNonce,
+        uint64 _auxData
+    )
+        external
+        payable
+        virtual
+        override(BridgeBase)
+        onlyRollup
+        returns (uint256 _outputValueA, uint256 _outputValueB, bool _interactionComplete)
+    {
+        // Get tokens from assets and aux data
+        (address _from, address _to) = _getTokensFromAuxData(_inputAssetA, _outputAssetA, _auxData);
+
+        // Terminate position and clean things up
+        uint256 _positionId = positionByNonce[_interactionNonce];
+        delete positionByNonce[_interactionNonce];
+        (uint256 _unswapped, uint256 _swapped) = DCA_HUB.terminate(_positionId, THIS_ADDRESS, THIS_ADDRESS);
+
+        if (_unswapped > 0) {
+            if (!DCA_HUB.paused()) {
+                // If there are still unswapped funds, then we will only allow closing the DCA position
+                // if swaps have been paused
```

Comment:
This is a limitation for shared positions, right? If we had 1 position p/user we ould allow this, right?

Reply:
We can't tell when that's the case, so users will only be able to terminate their position when it's finished